src/lib/CMakeLists.txt

```diff
@@ -244,6 +244,18 @@ set_target_properties(librnp
 
 if (BUILD_SHARED_LIBS)
   add_library(librnp-static STATIC $<TARGET_OBJECTS:librnp-obj> $<TARGET_OBJECTS:rnp-common>)
+  # Limit symbols export only to rnp_* functions.
+  if (APPLE)
+    # Check for Apple OSs and use -export_symbols_list
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/librnp.symbols" "_rnp_*\n")
+    target_link_options(librnp PRIVATE -Wl,-exported_symbols_list "${CMAKE_CURRENT_BINARY_DIR}/librnp.symbols")
+    set_target_properties(librnp PROPERTIES LINK_DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/librnp.symbols")
+  elseif(NOT WIN32)
+    # Use --version-script otherwise
+    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/librnp.vsc" "{\nglobal: rnp_*;\nlocal: *;\n};\n")
+    target_link_options(librnp PRIVATE -Wl,-O1 -Wl,--version-script=${CMAKE_CURRENT_BINARY_DIR}/librnp.vsc)
```

Comment:
Why does this change add `-O1` to the linker?  is that necessary for this fix?  it seems problematic because this might override other explicit choices for optimization (e.g. default settings in other build systems)

Reply:
It's not necessary, just noticed the thing below, inspired by the Ulrich Drepper's article on shared libs, part regarding the relocation speedups. However if it may cause issues it should be removed...
```
[root@ee62582d45c9 rnp-build]# readelf -I src/lib/librnp.so

Histogram for bucket list length (total of 1017 buckets):
 Length  Number     % of total  Coverage
      0  577        ( 56.7%)
      1  332        ( 32.6%)     59.0%
      2  94         (  9.2%)     92.4%
      3  13         (  1.3%)     99.3%
      4  1          (  0.1%)    100.0%

Histogram for `.gnu.hash' bucket list length (total of 421 buckets):
 Length  Number     % of total  Coverage
      0  230        ( 54.6%)
      1  154        ( 36.6%)     66.4%
      2  33         (  7.8%)     94.8%
      3  4          (  1.0%)    100.0%
[root@ee62582d45c9 rnp-build]# readelf -I /usr/local/lib   
lib/     lib64/   libexec/ 
[root@ee62582d45c9 rnp-build]# readelf -I /usr/local/lib64/librnp.so

Histogram for bucket list length (total of 263 buckets):
 Length  Number     % of total  Coverage
      0  43         ( 16.3%)
      1  78         ( 29.7%)     16.5%
      2  75         ( 28.5%)     48.3%
      3  40         ( 15.2%)     73.7%
      4  15         (  5.7%)     86.4%
      5  9          (  3.4%)     96.0%
      6  2          (  0.8%)     98.5%
      7  1          (  0.4%)    100.0%

Histogram for `.gnu.hash' bucket list length (total of 197 buckets):
 Length  Number     % of total  Coverage
      0  64         ( 32.5%)
      1  68         ( 34.5%)     29.2%
      2  41         ( 20.8%)     64.4%
      3  19         (  9.6%)     88.8%
      4  1          (  0.5%)     90.6%
      5  3          (  1.5%)     97.0%
      6  0          (  0.0%)     97.0%
      7  1          (  0.5%)    100.0%
```